src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -387,6 +410,9 @@ internal DocumentBranch(SourceText text, DocumentState state)
             }
         }
 
+        // This number is arbitrary. It is intended to keep the chain of incremental parses relatively small
+        private const int MAX_INCREMENTAL_PARSE_DEPTH = 32;
```

Comment:
I would like this to be an option, and not hardcoded in. That way we can adjust things without delivering a new binary if there's a problem.